src/OpenSSL/crypto.py

```diff
@@ -1765,6 +1772,47 @@ def verify_certificate(self):
         if ret <= 0:
             raise self._exception_from_context()
 
+    def get_verified_chain(self):
+        """
+        Verify a certificate in a context and return the complete validated
+        chain.
+
+        :raises X509StoreContextError: If an error occurred when validating a
+          certificate in the context. Sets ``certificate`` attribute to
+          indicate which certificate caused the error.
+
+        .. versionadded:: 20.0
+        """
+        # Always re-initialize the store context in case
+        # :meth:`verify_certificate` is called multiple times.
+        #
+        # :meth:`_init` is called in :meth:`__init__` so _cleanup is called
+        # before _init to ensure memory is not leaked.
+        self._cleanup()
```

Comment:
What happens if you call verify_certificate multiple times without reinitialization?

Reply:
Okay. This feels inelegant but that describes all of pyOpenSSL. I suspect if I asked why not just never init until you need it that I'd find out these objects can be passed to other parts of pyOpenSSL which emphatically expect them to have been init'd already.